### PR TITLE
Adding Dialog Fragment

### PR DESCRIPTION
**Pull request by julian-mainardis-globant**
## Description

Added a new fragment to set the number of parking available

## Story Type

- New feature :white_check_mark:
- Change request
- Bug fix
- Tech

## How Can This Be Tested?

Pressing the button in the main activity an then in the input, put the number of parking available, and then press "Aceptar"

## Evidences

| First Header  | Second Header |
| ------------- | ------------- |
| pixel2 | nexusS |

## Tested on

- Pixel 2
- Nexus S

## Package Structure
image

### CODE REVIEW

**Review by jnicolaslaurenti — APPROVED** [review]

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java [hunk 1]
```diff
@@ -4,15 +4,15 @@
 
 public class ParkingModel implements ParkingContract.Model {
 
-    private int parkingAvailable = -1;
+    private String parkingAvailable = "-1";
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: Here you can use a constant (PARKING_AVAILABLE_DEFAULT_VALUE for example). And it should be an int, why did you change it as string?

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
 
     @Override
-    public void setParkingAvailable(int parkingAvailable) {
+    public void setParkingAvailable(String parkingAvailable) {
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: Why did you change the int value (that was correct) for an string?

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java [hunk 3]
```diff
@@ -4,15 +4,15 @@
...
         this.parkingAvailable = parkingAvailable;
     }
 
     @Override
-    public int getParkingAvailable() {
+    public String getParkingAvailable() {
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: Same here

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/FragmentView.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.mobile.tandil.javabaseproject.mvp.view.base;
+
+import android.app.Fragment;
+import android.content.Context;
+import androidx.annotation.Nullable;
+
```
Comment (Georgina-Fernandez) [inline app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/FragmentView.java]: Erase this empty space

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java (hunk 1 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: because I was receiving it as a string from the fragment and I didn't want to parse it, but i already fixed

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/FragmentView.java (hunk 4 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/view/base/FragmentView.java]: fixed

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java (hunk 3 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java (hunk 2 above)
Comment (julian-mainardis-globant) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: done

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java (hunk 1 above)
Comment (Georgina-Fernandez) [reply app/src/main/java/com/mobile/tandil/javabaseproject/mvp/model/ParkingModel.java]: As we talk, we leave this as int in the model (because is the **number** of available lots and maybe we are going to need to perform operations with this value), and let the presenter to do the casts we need for both, the model and the view.

**Review by Georgina-Fernandez — APPROVED** [review]
Great job!

**Review by DiegoDiazGlobant — APPROVED** [review]

**Review by mariano98montero — APPROVED** [review]
